src/app/policies/components/PolicyConnections.vue

```diff
@@ -1,49 +1,60 @@
 <template>
-  <StatusInfo
-    :has-error="hasError"
-    :is-loading="isLoading"
-    :is-empty="!hasDataplanes"
+  <h2>Dataplanes</h2>
+
+  <input
+    id="dataplane-search"
+    v-model="searchInput"
+    type="text"
+    class="k-input mt-4"
+    placeholder="Filter by name"
+    required
+    data-testid="dataplane-search-input"
   >
-    <h2>Dataplanes</h2>
 
-    <input
-      id="dataplane-search"
-      v-model="searchInput"
-      type="text"
-      class="k-input mt-4"
-      placeholder="Filter by name"
-      required
-      data-testid="dataplane-search-input"
-    >
+  <DataSource
+    v-slot="{ data, isLoading, error }: PolicyDataplaneCollectionSource"
+    :src="`/${props.mesh}/${props.policyPath}/${props.policyName}/dataplanes`"
+  >
+    <LoadingBlock v-if="isLoading" />
+
+    <ErrorBlock
+      v-else-if="error"
+      :error="error"
+    />
 
-    <p
-      v-for="(dataplane, key) in filteredDataplanes"
-      :key="key"
-      class="mt-2"
-      data-testid="dataplane-name"
-    >
-      <router-link
-        :to="{
-          name: 'data-plane-detail-view',
-          params: {
-            mesh: dataplane.dataplane.mesh,
-            dataPlane: dataplane.dataplane.name,
-          },
-        }"
+    <EmptyBlock v-else-if="data === undefined || data.items.length === 0" />
```

Comment:
If I'm not mistaken these kinds of extra conditionals shouldn't be needed, if data is `undefined` then we are in a loading state. This fact is now hidden by the addition of `isLoading` whereas if we always use `typeof data === undefined` to check whether we are waiting for our initial load or not it becomes much clearer that these sorts of double conditionals aren't required.